internal/operator/nodedaemon_resource.go

```diff
@@ -0,0 +1,154 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func (r *KSANStorageReconciler) createNodeDaemon(ctx context.Context, namespace string) error {
+	podImage, err := podImage(ctx, r.Client, namespace)
+	if err != nil {
+		return err
+	}
+
+	resourceRequirements := corev1.ResourceRequirements{
+		Requests: corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse(nodeDaemonCPURequest),
+			corev1.ResourceMemory: resource.MustParse(nodeDaemonMemRequest),
+		},
+	}
+
+	labels := map[string]string{
+		"app.kubernetes.io/name":       "nodedaemon",
```

Comment:
These all would be better as global constants so that we can change them later on from a single place.